### PR TITLE
potential fix for #80

### PR DESCRIPTION
**Pull request by 0x53A**
ref https://github.com/mbraceproject/FsPickler/issues/80

I'm not sure if you even want to enable that scenario, since it may introduce non-determinism, but I think it would make sense ..

### CODE REVIEW

**Comment by dsyme** [comment]
@eiriktsarpalis Closing out old issues and PRs to keep the repo in good shape - what do you think of this one?

**Comment by 0x53A** [comment]
For context, I am currently running this as a fork in production, otherwise it is impossible (or at least I didn't find another way) to correctly serialize akka.net objects.

The pr was mostly meant as a starting point for a discussion.

**Comment by dsyme** [comment]
@eiriktsarpalis Can we accept this? I approve it but my understanding of the code is not that good

File: src/FsPickler/PicklerGeneration/PicklerResolution.fs [hunk 1]
```diff
@@ -33,19 +33,33 @@ let resolvePickler (resolver : IPicklerResolver)
...
+        
+        // get base types and implemented interfaces
+        let baseAndInterfaceTypes = [
+            let rec getBaseTypes (t : Type) = seq {
+                if t.BaseType <> null then
+                    yield t.BaseType
+                    yield! getBaseTypes t.BaseType
+            }
+            yield! getBaseTypes t
+            yield! t.GetInterfaces()
```
Comment (eiriktsarpalis) [inline src/FsPickler/PicklerGeneration/PicklerResolution.fs]: NB interfaces can also have base types

**Comment by eiriktsarpalis** [comment]
@dsyme Please see my comments in #80

File: src/FsPickler/PicklerGeneration/PicklerResolution.fs (hunk 1 above)
Comment (0x53A) [reply src/FsPickler/PicklerGeneration/PicklerResolution.fs]: As far as I know, ``GetInterfaces`` flattens the hierarchie. Simple test:

```F#
type I = interface end
type I2 = interface end

type I3 =
    inherit I
    inherit I2
	
Type A() =
    interface I
    interface I2
	
type B() =
    inherit A()
    interface I3
```

```F#
(new B()).GetType().GetInterfaces() |> Array.Map(fun t -> t.Name)

> val it : string [] = [|"I2"; "I"; "I3"|]
```

**Comment by eiriktsarpalis** [comment]
Fixed in FsPickler 4.0

**Comment by 0x53A** [comment]
Thanks!

I don't want to be too annoying, but nuget release when? ;-)

**Comment by eiriktsarpalis** [comment]
Just released

**Pull request closed without merging**